caf-audit/src/main/java/com/hpe/caf/auditing/webserviceclient/WebserviceClientAuditEventBuilder.java

```diff
@@ -0,0 +1,273 @@
+            LOG.error(errorMessage);
+            throw new IllegalArgumentException(errorMessage);
+        }
+
+        auditEventCommonFields.put(ElasticAuditConstants.FixedFieldName.TENANT_ID_FIELD, tenantId.toLowerCase());
+    }
+
+    @Override
+    public void setCorrelationId(String correlationId) {
+        auditEventCommonFields.put(ElasticAuditConstants.FixedFieldName.CORRELATION_ID_FIELD, correlationId);
+    }
+
+    @Override
+    public void setEventType(String eventCategoryId, String eventTypeId) {
+        auditEventCommonFields.put(ElasticAuditConstants.FixedFieldName.EVENT_CATEGORY_ID_FIELD, eventCategoryId);
+        auditEventCommonFields.put(ElasticAuditConstants.FixedFieldName.EVENT_TYPE_ID_FIELD, eventTypeId);
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, String value) {
+        auditEventParams.add(new EventParam(name, "string", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, short value) {
+        auditEventParams.add(new EventParam(name, "short", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, int value) {
+        auditEventParams.add(new EventParam(name, "int", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, long value) {
+        auditEventParams.add(new EventParam(name, "long", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, float value) {
+        auditEventParams.add(new EventParam(name, "float", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, double value) {
+        auditEventParams.add(new EventParam(name, "double", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, boolean value) {
+        auditEventParams.add(new EventParam(name, "boolean", columnName, value));
+    }
+
+    @Override
+    public void addEventParameter(String name, String columnName, Date value) {
+        auditEventParams.add(new EventParam(name, "date", columnName, value));
+    }
+
+    @Override
+    public void send() throws Exception {
+
+        //  Get the constructed Audit Event as a JSON string
+        String auditEventJson = getAuditEventAsJsonString();
+
+        // Reset the JsonBuilder
+        jsonBuilder = XContentFactory.jsonBuilder();
+
+        //  An Audit Event in Json format was not returned, throw exception
+        if (auditEventJson == null || auditEventJson.isEmpty()) {
+            String errorMessage = "No Audit Event JSON to send to the Audit webservice";
+            LOG.error(errorMessage);
+            throw new WebserviceClientException(errorMessage);
+        }
+
+        URL url = new URL("http://" + webserviceHostAndPort + "/caf-audit-service/v1/auditevents");
```

Comment:
This line will need changed in the future if we ever modify audit web service contract version.